agent_memory_server/vectorstore_adapter.py

```diff
@@ -891,6 +1112,47 @@ def parse_timestamp_to_datetime(timestamp_val):
             if len(memory_results) >= limit:
                 break
 
+        # Optional server-side recency-aware rerank (adapter-level fallback)
```

Comment:
Why is this here? We're already doing a database-level ranking if this parameter is passed. This is the fallback logic for the langchain adapter.